src/composables/fetch-stats-composable.ts

```diff
@@ -0,0 +1,36 @@
+import { type Ref, onMounted, onUnmounted, ref } from "vue";
+import { delay } from "@/libs/utils";
+import type { Granularity, XapiStatsResponse } from "@/libs/xapi-stats";
+import { useXenApiStore } from "@/stores/xen-api.store";
+
+export default function useFetchStats<T>(
+  type: "host" | "vm",
+  id: string,
+  granularity: Granularity
+) {
+  const stats = ref();
+
+  const xapiStats = useXenApiStore().getXapiStats();
+  const fetch = type === "host" ? xapiStats.getHostStats : xapiStats.getVmStats;
+
+  let watch = false;
+
+  onMounted(async () => {
+    watch = true;
+    while (watch) {
+      await fetchStats();
+    }
+  });
+
+  onUnmounted(() => {
+    watch = false;
+  });
+
+  const fetchStats = async () => {
+    const _stats = await fetch(id, granularity);
+    stats.value = _stats;
+    await delay(_stats.interval * 1000);
+  };
+
+  return { stats } as { stats: Ref<XapiStatsResponse<T>> };
+}
```

Comment:
Je pense que ça peut être simplifié avec quelque chose de ce genre (pas testé):

```suggestion
import { promiseTimeout, useIntervalFn } from '@vueuse/core';
import type { Granularity, XapiStatsResponse } from "@/libs/xapi-stats";
import { useXenApiStore } from "@/stores/xen-api.store";

export default function useFetchStats<T>(
  type: "host" | "vm",
  id: string,
  granularity: Granularity
) {
  const stats = ref<XapiStatsResponse<T>>();
  const xapiStats = useXenApiStore().getXapiStats();
  const fetch = type === "host" ? xapiStats.getHostStats : xapiStats.getVmStats;
  
  const fetchStats = async () => {
    stats.value = await fetch(id, granularity);
    await promiseTimeout(stats.value.interval * 1000);
  }

  useIntervalFn(fetchStats);
  
  return { stats };
}
```